xelAPI/src/matches/lolMatch.service.ts

```diff
@@ -0,0 +1,178 @@
+        'MatchSchedule.BestOf',
+        'MatchSchedule.ShownName',
+        'MatchSchedule._ID',
+      ],
+      where: whereClause,
+      orderBy: [
+        {
+          field: 'MatchSchedule.DateTime_UTC',
+          desc: true,
+        },
+      ],
+    });
+
+    const matchesObj = matches.data;
+    matchesObj.forEach(match => {
+      match['gameId'] = 1;
+    });
+
+    return matchesObj;
+  }
+
+  /**
+ * Retrieve League of Legends matches for specific teams within the last three weeks.
+ *
+ * @param teamNames - An array of team names for which to retrieve matches.
+ * @returns An array of League of Legends matches for the specified teams.
+ */
+  async getLolTeamsMatches(teamNames: string[]) {
+    if (teamNames.length === 0) {
+      // If the array is empty, return an empty result.
+      return [];
+    }
+
+    // Construct the WHERE clause for all specified team names.
+    const whereClause = `MatchSchedule.DateTime_UTC >= DATE("${this.getThreeWeeksAgoISO()}") AND (` +
+      teamNames.map(teamName => `MatchSchedule.Team1="${teamName}" OR MatchSchedule.Team2="${teamName}"`).join(" OR ") +
+      ")";
+
+    const teamsMatches = await this.queryMatchesFromCargo(whereClause);
+    console.log(teamsMatches)
+    if (teamsMatches && teamsMatches.length > 0) {
+      return teamsMatches;
+    } else {
+      throw new Error("No matches found for the specified teams");
+    }
+  }
+
+   /**
+ * Retrieve League of Legends matches for specific leagues within the last three weeks.
+ *
+ * @param leaguesNames - An array of team leagues for which to retrieve matches.
+ * @returns An array of League of Legends matches for the specified leagues.
+ */
+   async getLolLeaguesMatches(leaguesNames: string[]) {
+    if (leaguesNames.length === 0) {
+      // If the array is empty, return an empty result.
+      return [];
+    }
+
+    // Construct the WHERE clause for all specified leagues names.
+    const whereClause = `MatchSchedule.DateTime_UTC >= DATE("${this.getThreeWeeksAgoISO()}") AND (` +
+    leaguesNames.map(leagueName => `MatchSchedule.ShownName LIKE "%${leagueName}%"`).join(" OR ") +
+    ")";
+    const leaguesMatches = await this.queryMatchesFromCargo(whereClause);
+    if (leaguesMatches && leaguesMatches.length > 0) {
+      return leaguesMatches;
+    } else {
+      throw new Error("No matches found for the specified leagues");
+    }
+  }
+
+
+  /**
+   * Retrieve all League of Legends matches within the last three weeks.
+   *
+   * @returns An array of all League of Legends matches within the last three weeks.
+   */
+  async getAllLolmMatches() {
+    return this.queryMatchesFromCargo(
+      `MatchSchedule.DateTime_UTC >= DATE("${this.getThreeWeeksAgoISO()}")`
+    );
+  }
+
+  /**
+   * Get the ISO representation of the date three weeks ago from the current date.
+   *
+   * @returns An ISO date string for three weeks ago from the current date.
+   */
+  private getThreeWeeksAgoISO() {
```

Comment:
tu devrais faire de constante de classe au lieu de 21 en dur comme ca tu pourra changer plus facilement et pas tout renommer dans les commentaire des précédentes fonctions 

et pour le dev comme ca je te conseille de faire des services boite à outils pour pouvoir les réutiliser à la volée

Reply:
C'est noté merci je vais essayer de faire ça !